src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -2329,7 +2329,7 @@ void MacroAssembler::incrementl(Address dst, int value) {
 
 void MacroAssembler::jump(AddressLiteral dst, Register rscratch) {
   assert(rscratch != noreg || always_reachable(dst), "missing");
-
+  assert(!dst.rspec().reloc()->is_data(), "should not use ExternalAddress for jump");
```

Comment:
Should these assert be added for other platforms as well?

Reply:
It is not easy for me to find instructions in other ports into which to add such assert. For example, aarch64 use several instructions to load address for branch. I hope engineers responsible for ports will add such assert where it is needed.